iceoryx_hoofs/buffer/include/iox/detail/stack.inl

```diff
@@ -144,10 +141,9 @@ inline optional<T> stack<T, Capacity>::pop() noexcept
         return nullopt;
     }
 
-    // AXIVION Next Construct AutosarC++19_03-A5.2.4 : low level memory management with access to the topmost element on
-    // the untyped buffer; reinterpret_cast is safe since the size and the alignment of each array element is guaranteed
-    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
-    return *reinterpret_cast<T*>(&m_data[--m_size]);
+    optional<T> element{std::move(m_data[--m_size])};
```

Comment:
Mmmhh does it mean we loosen the type requirements here a bit so that types only have to be move constructible?

Reply:
@elfenpiff Yep, that's correct. I added the `static_assert`.